### ISSUE TITLE
Run travis on OSX as well

### ISSUE THREAD

**Issue by hchauvin** [grailbio/rules_r#26]
I think it was not available in the past, but now it is possible to run travis on OSX for Open Source projects.

**Comment by siddharthab** [comment]
The main reason I did not enable this before was because I thought we won't be able to use container based builds in linux. Let me test this out and see what the hit is.

**Comment by siddharthab** [comment]
I [tried](https://travis-ci.org/siddharthab/rules_r/jobs/375655233).

The linux build is still based on containers, so that concern is taken care of. But macOS is failing at one point. Because the macOS setup is VM based, I can not really reproduce the environment and see what is going wrong. The tests pass on my mac.

**Comment by hchauvin** [comment]
Thanks for trying! I haven't been able to identity the source of the failure either, but it somehow disappears with R-3.5.0 and was not present with R-3.3.0.

**Comment by siddharthab** [comment]
I tried R 3.5.0 as well, but I am having other unrelated problems there -- https://travis-ci.org/siddharthab/rules_r/jobs/377939301. Other users have reported seeing the same issue with R 3.5.0.

**Comment by hchauvin** [comment]
The message you see concerning the compilation of rlang comes from the fact that R-3.5.0 checks for an internal ID in Meta/features.rds for packages that contain compiled code. If Meta/features.rds is absent or the internal ID is different, this error is raised.

You need to do something like that in _package_files in build.bzl:
```
pkg_files += [
    ctx.actions.declare_file("lib/{0}/Meta/features.rds".format(pkg_name)),
]
```

See https://travis-ci.org/hchauvin/rules_r/builds/378047858.

It makes me think that there should probably be a test in './tests' to make sure that nothing unexpected is created and not registered as an output when building a package, because this kind of behavior is extremely hard to debug.

**Comment by siddharthab** [comment]
Got it! Thanks for looking into it. I have to give a little more time to understand when features.rds is expected and when not. I tried to do that in the very early days, but could not find a reliable marker and so left it out of pkg_files.

In general, I have to go through all the packages I have and check the output of file_list_diff to see which files we miss collecting from the sandbox.

**Comment by siddharthab** [comment]
This is now working with https://github.com/grailbio/rules_r/commit/bb6532e517175bf2f01fa2a8e14545f1c5f7025c. I will get this pushed through our internal system.

**Comment by siddharthab** [comment]
Resolved by c68c7b3ee0d783555b8cd1f1e706aba1aeaca155.
